[IMP] Website: review website translation page (#3973)
odoo/documentation: content/applications/websites/website/publish/translate.rst

===================
Website translation
===================

The contents of your website pages (i.e. text strings) can be translated into different languages
directly on your website.

.. note::
   Your website is displayed in the language that matches the visitor's browser's language, unless
   that particular language has not been installed. In this case, the website is displayed in the
   :ref:`default language <default-language>`. The visitor can still select another language in
   the language menu.

Installing languages
====================

To translate your website, you first have to add the required language(s):

#. Go to your website.
#. Scroll at the bottom of the page to the **language menu**.
#. Click the language and select :guilabel:`Add a language`.

   .. image:: translate/website-add-language.png
      :alt: Add a language to your website.

#. Click the :guilabel:`Languages` field and select the required language from the drop-down
   list. Repeat this step for each additional language.
#. Click the :guilabel:`Add` button.

.. tip::
   - You can also edit your website's languages from the backend, in the :guilabel:`Settings`. Go to

>      :menuselection:`Website –> Configuration –> Settings`, and add/remove the required

Reviewer: I would add "and add/remove the required language in the Website info section" or something like this, as the page is quite long.

     languages in the :guilabel:`Languages` field.

   .. _default-language:

   - The website's default language can be defined in the :guilabel:`Default` field in the

>      :guilabel:`Settings`.

Reviewer: Same as above, I would suggest to mention the section name (Website info).

Translating the contents
========================

Once the required languages have been added, you can translate the contents of your website. To do

> so, go to your website, select the required language from the language menu and click the

Reviewer: This is very nitpicky but "required language" sounds a bit like it's a language required by the system. Perhaps "once a new language has been added, you can translate the contents of your website (...). Select the language from the menu and (...).

> :guilabel:`Translate` button in the upper right part to activate the **Translation mode**.

Reviewer: Perhaps "on the right side of the upper taskbar" or "on the right side of the taskbar at the top"? "Part" seems a bit vague.

.. image:: translate/translate-button.png
   :alt: Translate button

As a result:

- Text strings that have already been translated are highlighted in green;
- Text strings that need to be translated are highlighted in yellow.

.. image:: translate/website-translation-yellow.png
   :align: center
   :alt: Text to be translated highlighted in yellow

> You can then replace the original text with the translation directly on your website.

Reviewer: For clarity, you could add "by clicking on a text block and editing it."

.. tip::
   - Once the required languages have been installed, you can also translate some items (e.g.,
     the product's name and description) from the backend (e.g., in the product template). To do so,
     click the language code (e.g., :guilabel:`EN`) next to the text you want to translate (e.g.,
     the product name) and edit the required translation.

> 

Reviewer: It was not super clear to me where to go, I thought it would be in backend of Website app, but I was only able to find it in Sales. If it's possible in Website, maybe add more details about where, if it's only in the concerned app, then I'd add this to clarify.

Idk how the examples are normally handled, but perhaps it's a good option to use the example markup and add it under the instructions?

===================
Website translation
===================

> The contents of your website pages (i.e. text strings) can be translated into different languages

Reviewer 2: ```suggestion
The contents of your website pages (i.e., text strings) can be translated into different languages
```

directly on your website.

.. note::

>    Your website is displayed in the language that matches the visitor's browser's language, unless

Reviewer 2: ```suggestion
   Your website is displayed in the language that matches the visitor's browser's language unless
```

   that particular language has not been installed. In this case, the website is displayed in the

>    :ref:`default language <default-language>`. The visitor can still select another language in
>    the language menu.

Reviewer 2: nitpicking
```suggestion
   :ref:`default language <default-language>`. The visitor can still select another language in the
   language menu.
```

Installing languages
====================

> To translate your website, you first have to add the required language(s):

Reviewer 2: ```suggestion
To translate your website, you first have to add the required languages:
```

#. Go to your website.
#. Scroll to the bottom of the page to the **language menu**.
#. Click the language and select :guilabel:`Add a language`.

   .. image:: translate/website-add-language.png
      :alt: Add a language to your website.

> #. Click the :guilabel:`Languages` field and select the required language from the drop-down
>    list. Repeat this step for each additional language.

Reviewer 2: More nitpicking. I think it's because of your automatic hard wrap on PyCharm
```suggestion
#. Click the :guilabel:`Languages` field and select the required language from the drop-down list.
   Repeat this step for each additional language.
```

#. Click the :guilabel:`Add` button.

.. tip::

>    - You can also edit your website's languages from the backend, in the :guilabel:`Settings`. Go to
>      :menuselection:`Website –> Configuration –> Settings`, and add/remove the required
>      languages in the :guilabel:`Languages` field, in the :guilabel:`Website info` section.

Reviewer 2: ```suggestion
   - You can also edit your website's languages from the backend in the :guilabel:`Settings`. Go to
     :menuselection:`Website –> Configuration –> Settings`, and add/remove the required
     languages in the :guilabel:`Languages` field in the :guilabel:`Website info` section.
```

>    .. _default-language:

Reviewer 2: I'm unsure about putting a custom anchor in the middle of a list... Let's see together during the IRL review.
also, I suggest writing anchors like this `translate/default-language`

   - The website's default language can be defined in the :guilabel:`Default` field in the

>      :guilabel:`Settings`.

Reviewer 2: I think it's mostly because this content is not a tip. It is a flow of its own, even if it is an easy one.

Translating the contents
========================

> Once the languages have been added, you can translate the contents of your website. To do
> so, go to your website, select the language from the language menu and click the
> :guilabel:`Translate` button on the right part of the task bar to activate the **Translation
> mode**.

Reviewer 2: even more 100 characters nitpicking + commas + "taskbar" 😁
```suggestion
Once the languages have been added, you can translate the contents of your website. To do so, go to
your website, select the language from the language menu, and click the :guilabel:`Translate` button
on the right part of the taskbar to activate the **Translation mode**.
```

.. image:: translate/translate-button.png
   :alt: Translate button

As a result:

- Text strings that have already been translated are highlighted in green;
- Text strings that need to be translated are highlighted in yellow.

.. image:: translate/website-translation-yellow.png
   :alt: Text to be translated highlighted in yellow

You can then replace the original text with the translation by clicking the block and editing

> it.

Reviewer 2: + save

.. tip::
   - Once the languages have been installed, you can also translate some items (e.g.,
     the product's name and description) from the backend (e.g., in the product template). To do so,
     click the language code (e.g., :guilabel:`EN`) next to the text you want to translate (e.g.,
     the product name) and add the translation.

     .. image:: translate/product-translation.png
        :alt: Translate product-related items.

   - You can also :doc:`export/import translations <../../../../developer/howtos/translations>`
     to translate multiple items (e.g., product names and descriptions) in one go.

Adding a language selector menu

> ===============================

Reviewer 2: "Adding" doesn't add anything to the info.
